### ISSUE TITLE
Won't install for me (under ubuntu 14.04.3)

### ISSUE THREAD

**Issue by draxil** [cblanc/hiveline#1]
Won't install for me (under ubuntu 14.04.3)

> $ sudo gem install hiveline
> Building native extensions.  This could take a while...
> ERROR:  Error installing hiveline:
>        ERROR: Failed to build gem native extension.
> 
> ```
>    /usr/bin/ruby1.9.1 extconf.rb
> ```
> 
> /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require': cannot load such file -- mkmf (LoadError)
>        from /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require'
>        from extconf.rb:1:in `<main>'
> Gem files will remain installed in /var/lib/gems/1.9.1/gems/json-1.8.3 for inspection.
> Results logged to /var/lib/gems/1.9.1/gems/json-1.8.3/ext/json/ext/generator/gem_make.out

**Comment by cblanc** [comment]
So I was able to reproduce this on a clean vm running ubuntu 14.04 but with ruby 1.9.1

I found a whole bunch of stackoverflow questions and github issues related to ruby 1.9.1 not building json v1.8.3 properly

It seems this is a ruby 1.9.x issue. In the end, the solution that worked for me was to upgrade ruby to 1.9.3 with `sudo apt-get install ruby-dev`. `gem install json -v 1.8.3` subsequently worked and the hiveline gem installed with `sudo gem install hiveline`

While testing on ubuntu I came across another issue, ruby 1.9.1 doesn't play well with utf-8 out of the box. I think I've fixed this issue just now with a patch, new version is 0.0.7
